/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_set_quality@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_quality_scaling@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_destroy_decompress@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_resync_to_restart@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_CreateDecompress@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_alloc_huff_table@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_set_defaults@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_CreateCompress@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_stdio_src@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_finish_compress@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_calc_output_dimensions@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_std_error@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_default_qtables@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_start_decompress@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_write_scanlines@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_simple_progression@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_read_header@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_read_scanlines@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_start_compress@LIBJPEG_9.0’未定义的引用
/home/lyc/anaconda2/lib/libopencv_imgcodecs.so：对‘jpeg_destroy_compress@LIBJPEG_9.0’未定义的引用



发现是找不到libpng16.so.16与libjpeg.so.9文件．打开anaconda中的…anaconda2/lib路径，发现anaconda下有libpng16.so.16与libjpeg.so.9，于是进行如下操作：

cd /usr/lib/x86_64-linux-gnu
sudo ln -s ~/anaconda2/lib/libpng16.so.16 /usr/lib/
sudo ln -s ~/anaconda2/lib/libjpeg.so.9 /usr/lib/
sudo ldconfig
1
2
3
4
为这两个文件在/usr/lib/下建立一个同步的链接． 
再次make，问题消失．
--------------------- 
作者：阿卡蒂奥 
来源：CSDN 
原文：https://blog.csdn.net/akadiao/article/details/79834086 
版权声明：本文为博主原创文章，转载请附上博文链接！
